src/formatDate.js

```diff
@@ -1,14 +1,54 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prevSplitter = fromFormat[3];
+  const newSplitter = toFormat[3];
```

Comment:
Similar to the previous comment, this line assumes the separator is at the third index of the format string. Ensure that the separator is correctly identified for all possible formats.